### PR TITLE
command switch can now take more complicated arguments.

### PR DESCRIPTION
**Pull request by mohsinhijazee**
The --command switch now can take more complicated arguments like this:

```
ecs deploy my-cluster my-service --command my-container '["sh", "-c", "while true; do echo Time files like an arrow $(date); sleep 1; done;"]'
```

`README` and tests are updated.

Please take a look @fabfuel

### CODE REVIEW

File: ecs_deploy/ecs.py [hunk 1]
```diff
@@ -591,3 +611,6 @@ class TaskPlacementError(EcsError):
 
 class UnknownTaskDefinitionError(EcsError):
     pass
+
```
Comment (fabfuel) [inline ecs_deploy/ecs.py]: Please add the missing empty line here (PEP8)

File: ecs_deploy/ecs.py [hunk 2]
```diff
@@ -191,8 +200,19 @@ def get_overrides(self):
         return overrides
 
     @staticmethod
-    def get_overrides_command(command):
+    def parse_command(command):
+        if re.match(JSON_LIST_REGEX, command):
+            try:
+                return json.loads(command)
+            except JSONDecodeError as e:
+                raise EcsTaskDefinitionCommandError("command should be valid JSON list."
```
[the remark this answers is not in the document]
Comment (mohsinhijazee) [reply ecs_deploy/ecs.py]: > The lines are concatenated without whitespace, resulting in:
> `command should be valid JSON list.Got following command: ["foo, "bar"]Resulting in error: Expecting ',' delimiter: line 1 column 9 (char 8)`
> 
> Please reformat this and please try to stay within 79 chars line-length (PEP8).

Thank you for pointing out, I have updated and the line length is now within 79 characters. BTW I note that on few places, the line length is >79 so if you want, I can submit a PR to fix all the PEP8.

File: ecs_deploy/ecs.py [hunk 3]
```diff
@@ -191,8 +200,19 @@ def get_overrides(self):
...
+                                                    "Got following command: {}"
+                                                    "Resulting in error: {}".format(command, str(e)))
+
         return command.split(' ')
+    @staticmethod
```
[the remark this answers is not in the document]
Comment (mohsinhijazee) [reply ecs_deploy/ecs.py]: Fixed

File: ecs_deploy/ecs.py (hunk 1 above)
Comment (mohsinhijazee) [reply ecs_deploy/ecs.py]: Fixed.

File: ecs_deploy/ecs.py [hunk 4]
```diff
@@ -1,9 +1,18 @@
...
 from botocore.exceptions import ClientError, NoCredentialsError
 from dateutil.tz.tz import tzlocal
 
+JSON_LIST_REGEX = re.compile(r'^\[.*\]$')
+
+# Python2 raises ValueError
+try:
+    JSONDecodeError = json.JSONDecodeError
+except AttributeError:
+    JSONDecodeError = ValueError
```
[the remark this answers is not in the document]
Comment (mohsinhijazee) [reply ecs_deploy/ecs.py]: Fixed, missing line is added.

**Comment by fabfuel** [comment]
Thanks @mohsinhijazee & sorry for the delay

**Pull request merged**